### PR TITLE
Add GitHub Action

### PR DESCRIPTION
**Pull request by lindexi**
See https://github.com/dotnet-campus/Open-XML-SDK/pull/1/checks?check_run_id=670319346

Fix https://github.com/OfficeDev/Open-XML-SDK/issues/666

All the developer can find the build and test log.

And the GitHub Action only be used in MR and not be used to publish nuget package that means it need not sign package.

### CODE REVIEW

**Comment by lindexi** [comment]
Sorry, I use the wrong repo

**Comment by twsouthwick** [comment]
I'm fine adding this but you'd want to build all targets. This requires msbuild and setting the variable `ProjectLoadStyle=All`. Check out the Readme for details

File: .github/workflows/Build.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      uses: actions/setup-dotnet@v1
+      with: 
+        dotnet-version: 1.1.14
+
+    - name: Setup dotnet 3.1
+      uses: actions/setup-dotnet@v1
+      with: 
+        dotnet-version: 3.1.202
+
+    - name: Setup dotnet 5.0
```
Comment (twsouthwick) [inline .github/workflows/Build.yml:23]: We don't use .NET 5 at the moment. I'd rather not use prerelease if possible

File: .github/workflows/Build.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    - name: Setup NuGet
+      uses: nuget/setup-nuget@v1
+      with:
+        nuget-version: '5.x'
+
+    - name: Add msbuild to PATH
+      uses: microsoft/setup-msbuild@v1.0.0
+
+    - name: Add nuget source
+      run: dotnet nuget add source -n dotnet5 https://dnceng.pkgs.visualstudio.com/public/_packaging/dotnet5/nuget/v3/index.json
```
Comment (twsouthwick) [inline .github/workflows/Build.yml:37]: Why are you adding this?

**Comment by lindexi** [comment]
@twsouthwick I need your help. I find the DocumentFormat.OpenXml.Packaging.Tests.csproj and IsolatedStorageExceptionWorkaround.csproj and so on can not compile in NET35

```
"D:\a\Open-XML-SDK\Open-XML-SDK\Open-XML-SDK.sln" (default target) (1) ->
"D:\a\Open-XML-SDK\Open-XML-SDK\samples\IsolatedStorageExceptionWorkaround\IsolatedStorageExceptionWorkaround.csproj" (default target) (8) ->
"D:\a\Open-XML-SDK\Open-XML-SDK\samples\IsolatedStorageExceptionWorkaround\IsolatedStorageExceptionWorkaround.csproj" (Build target) (8:2) ->
  Program.cs(80,43): error CS0012: The type 'Package' is defined in an assembly that is not referenced. You must add a reference to assembly 'WindowsBase, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. [D:\a\Open-XML-SDK\Open-XML-SDK\samples\IsolatedStorageExceptionWorkaround\IsolatedStorageExceptionWorkaround.csproj]
  Program.cs(95,43): error CS0012: The type 'Package' is defined in an assembly that is not referenced. You must add a reference to assembly 'WindowsBase, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. [D:\a\Open-XML-SDK\Open-XML-SDK\samples\IsolatedStorageExceptionWorkaround\IsolatedStorageExceptionWorkaround.csproj]
```

File: .github/workflows/Build.yml (hunk 2 above)
Comment (lindexi) [reply .github/workflows/Build.yml]: @twsouthwick Sorry, this is the copy code from my other repo

**Comment by lindexi** [comment]
@twsouthwick The dotnet framework and the dotnet core 1.1 will only build the main project, because they can not compile the IsolatedStorageExceptionWorkaround.csproj and so on.

The test will only run in dotnet core 2.1

 What do you think?

**Comment by lindexi** [comment]
All the task can run parallel.



See: https://github.com/dotnet-campus/Open-XML-SDK/pull/1

https://github.com/dotnet-campus/Open-XML-SDK/pull/1/checks?check_run_id=692572748

**Comment by lindexi** [comment]
Why I do not use `msbuild /m /p:configuration="release" /p:ProjectLoadStyle=All` to compile?

Because it can not run successfully.

See https://github.com/dotnet-campus/Open-XML-SDK/runs/688931364?check_suite_focus=true

**Comment by twsouthwick** [comment]
Looks like there are some pre-reqs that are not installed. I'll take a look later and see what's missing

**Comment by twsouthwick** [comment]
I've pushed a change to allow building everything with dotnet rather than msbuild. Take a look at the status checks now - they may provide enough information that this isn't needed.

**Comment by twsouthwick** [comment]
The results now show helpful info via devops:

image

**Comment by lindexi** [comment]
@twsouthwick Thank you.  Waiting for your good news

**Comment by twsouthwick** [comment]
With the changes I mentioned above, I'm not sure this is still needed. Thoughts? I'd rather not have duplicated checks.

**Comment by lindexi** [comment]
@twsouthwick Do you means that the developer can find the Azure DevOps when build error?

But I do not have access.

**Comment by twsouthwick** [comment]
No, you won't be able to see that, but you should be able to see the checks 
 for instance at https://github.com/OfficeDev/Open-XML-SDK/pull/727/checks

**Comment by lindexi** [comment]
@twsouthwick Thank you. I know what you mean.

**Comment by lindexi** [comment]
@twsouthwick I want to close this PR, because we can use the Azure DevOps

**Pull request closed without merging**